easybuild/tools/systemtools.py

```diff
@@ -133,76 +103,107 @@ def get_core_count():
+        res = vendor_regex.search(txt)
+        if res:
+            arch = res.group(1) or res.group(2)
+        if arch in VENDORS:
+            vendor = VENDORS[arch]
+            tup = (vendor, vendor_regex.pattern, PROC_CPUINFO_FP)
+            _log.debug("Determined CPU vendor on Linux as being '%s' via regex '%s' in %s" % tup)
 
     elif os_type == DARWIN:
-        out, exitcode = run_cmd("sysctl -n machdep.cpu.vendor")
+        cmd = "sysctl -n machdep.cpu.vendor"
+        out, ec = run_cmd(cmd)
         out = out.strip()
-        if not exitcode and out and out in VENDORS:
-            return VENDORS[out]
+        if ec == 0 and out in VENDORS:
+            vendor = VENDORS[out]
+            _log.debug("Determined CPU vendor on DARWIN as being '%s' via cmd '%s" % (vendor, cmd))
+
+    if vendor is None:
+        vendor = UNKNOWN
+        _log.warning("Could not determine CPU vendor on %s, returning %s" % (os_type, vendor))
+
+    return vendor
+
+
+def get_cpu_family():
+    """
+    Determine CPU family.
+    @return: one of the AMD, ARM, INTEL, POWER constants
+    """
+    family = None
+    vendor = get_cpu_vendor()
+    if vendor in CPU_FAMILIES:
+        family = vendor
+        _log.debug("Using vendor as CPU family: %s" % family)
 
     else:
-        # BSD
-        out, exitcode = run_cmd("sysctl -n hw.model")
-        out = out.strip()
-        if not exitcode and out:
-            return out.split(' ')[0]
+        # POWER family needs to be determined indirectly via 'cpu' in /proc/cpuinfo
+        if os.path.exists(PROC_CPUINFO_FP):
+            cpuinfo_txt = read_file(PROC_CPUINFO_FP)
+            power_regex = re.compile(r"^cpu\s+:\s*POWER.*", re.M)
+            if power_regex.search(cpuinfo_txt):
+                family = POWER
+                tup = (power_regex.pattern, PROC_CPUINFO_FP, family)
+                _log.debug("Determined CPU family using regex '%s' in %s: %s" % tup)
+
+    if family is None:
+        family = UNKNOWN
+        _log.warning("Failed to determine CPU family, returning %s" % family)
 
-    return UNKNOWN
+    return family
 
 
 def get_cpu_model():
     """
-    returns cpu model
-    f.ex Intel(R) Core(TM) i5-2540M CPU @ 2.60GHz
+    Determine CPU model
+    for example: Intel(R) Core(TM) i5-2540M CPU @ 2.60GHz
     """
+    model = None
     os_type = get_os_type()
-    if os_type == LINUX:
-        regexp = re.compile(r"^model name\s+:\s*(?P<modelname>.+)\s*$", re.M)
-        try:
-            txt = read_file('/proc/cpuinfo', log_error=False)
-            if txt is not None:
-                return regexp.search(txt).groupdict()['modelname'].strip()
-        except IOError, err:
-            raise SystemToolsException("An error occured when determining CPU model: %s" % err)
+
+    if os_type == LINUX and os.path.exists(PROC_CPUINFO_FP):
+        # consider 'model name' first, use 'model' as a fallback
+        # 'model name' is not there for Linux/POWER, but 'model' has the right info
```

Comment:
no, usually both `model` and `model name` are there (and typically `model` is listed first)

if they're both there, we *must* use the `model name` entry, otherwise it's probably `model`